File: site-keeper-frontend/src/models/Site/SiteRequestModel.ts
import CostingModel from "../CostingModel"
import UserModel from "../User/UserModel"

class SiteRequestModel{
    id: number
    name: string
    abbreviation: string
    siteManager: UserModel
    logo?: string
    location: string
    betslipId: number

    constructor(id: number, name: string, siteManager: UserModel, logo: string,
        location: string, abbreviation: string, betslipId: number) {
        this.id = id
        this.name = name
        this.siteManager = siteManager
        this.logo = logo
        this.location = location
        this.abbreviation = abbreviation
        this.betslipId = betslipId
    }
}

export default SiteRequestModel

File: site-keeper-data-access/src/main/java/com/igt/sitekeeper/requestmodel/SiteRequestModel.java
package com.igt.sitekeeper.requestmodel;

import com.igt.sitekeeper.model.Costing;
import com.igt.sitekeeper.model.User;
import com.igt.sitekeeper.model.relationmodels.DisciplinesOnSites;
import com.igt.sitekeeper.model.relationmodels.TerminalsOnSites;
import com.igt.sitekeeper.requestmodel.relationrequestmodels.TerminalsOnSitesRequestModel;
import lombok.Data;

import java.util.List;

@Data
public class SiteRequestModel {

    private String name;
    private User siteManager;
    private String logo;
    private String location;
    private String abbreviation;
    private Double betslipId;
    private List<TerminalsOnSitesRequestModel> terminalsOnSites;
    private List<DisciplinesOnSites> disciplines;
}
